File: app/src/main/java/com/pij/zworkout/list/viewmodel/CreateWorkoutFeature.java
package com.pij.zworkout.list.viewmodel;

import android.support.annotation.NonNull;

import com.pij.horrocks.Interaction;
import com.pij.horrocks.Reducer;
import com.pij.zworkout.list.Model;

/**
 * <p>Created on 01/03/2018.</p>
 *
 * @author Pierrejean
 */

public class CreateWorkoutFeature implements Interaction<Object, Model> {
    @NonNull
    @Override
    public Reducer<Model> process(@NonNull Object event) {
        return current -> current.toBuilder()
                .createWorkout(true)
                .build();
    }
}

File: app/src/main/java/com/pij/zworkout/list/feature/CreateWorkoutFeature.kt
package com.pij.zworkout.list.feature

import com.pij.horrocks.Interaction
import com.pij.horrocks.Reducer
import com.pij.zworkout.list.Model

/**
 *
 * Created on 01/03/2018.
 *
 * @author Pierrejean
 */

class CreateWorkoutFeature : Interaction<Any, Model> {

    override fun process(event: Any): Reducer<Model> {
        return Reducer { it.copy(createWorkout = true) }
    }
}
